code/modules/mining/machine_silo.dm

```diff
@@ -96,6 +169,43 @@
 	balloon_alert(user, "saved to multitool buffer")
 	return ITEM_INTERACT_SUCCESS
 
+/obj/machinery/ore_silo/proc/connect_receptacle(datum/component/remote_materials/receptacle, atom/movable/physical_receptacle)
+	ore_connected_machines += receptacle
+	receptacle.mat_container = src.materials
+	receptacle.silo = src
+	RegisterSignal(physical_receptacle, COMSIG_ORE_SILO_PERMISSION_CHECKED, PROC_REF(check_permitted))
+
+/obj/machinery/ore_silo/proc/disconnect_receptacle(datum/component/remote_materials/receptacle, atom/movable/physical_receptacle)
+	ore_connected_machines -= receptacle
+	receptacle.mat_container = null
+	receptacle.silo = null
+	holds -= receptacle
+	UnregisterSignal(physical_receptacle, COMSIG_ORE_SILO_PERMISSION_CHECKED)
+
+/obj/machinery/ore_silo/proc/check_permitted(datum/source, alist/user_data, atom/movable/physical_receptacle)
+	SIGNAL_HANDLER
+
+	if(!ID_required)
+		return COMPONENT_ORE_SILO_ALLOW
+	if(!islist(user_data))
+		// Just allow to salvage the situation
+		. = COMPONENT_ORE_SILO_ALLOW
+		CRASH("Invalid data passed to check_permitted")
+	if(user_data[SILICON_OVERRIDE] || user_data[CHAMELEON_OVERRIDE] || astype(user_data["Accesses"], /list)?.Find(ACCESS_QM))
+		return COMPONENT_ORE_SILO_ALLOW
+	if(user_data[ID_READ_FAILURE])
+		physical_receptacle.say("SILO ERR: ID interface failure. Please contact the Head of Personnel.")
+		return COMPONENT_ORE_SILO_DENY
+	if(!user_data["Account ID"] || !isnum(user_data["Account ID"]))
+		if(prob(5))
+			physical_receptacle.say("SILO ERR: Bank account ID not found. Initiating anti-communist silo-access policy.")
+		physical_receptacle.say("SILO ERR: No account ID found. Please contact a banker.")
```

Comment:
Considering we don't have a job whose role really gets called a banker, I would just call out the Head of Personnel directly.